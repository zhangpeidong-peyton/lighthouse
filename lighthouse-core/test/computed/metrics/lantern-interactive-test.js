/**
 * @license Copyright 2017 The Lighthouse Authors. All Rights Reserved.
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */
'use strict';

const LanternInteractive = require('../../../computed/metrics/lantern-interactive.js');
const assert = require('assert').strict;

const trace = require('../../fixtures/traces/progressive-app-m60.json');
const devtoolsLog = require('../../fixtures/traces/progressive-app-m60.devtools.log.json');
const iframeTrace = require('../../fixtures/traces/iframe-m79.trace.json');
const iframeDevtoolsLog = require('../../fixtures/traces/iframe-m79.devtoolslog.json');

/* eslint-env jest */

describe('Metrics: Lantern TTI', () => {
  it('should compute predicted value', async () => {
    const settings = {};
    const context = {settings, computedCache: new Map()};
    const result = await LanternInteractive.request({trace, devtoolsLog,
      settings}, context);

    expect({
      timing: Math.round(result.timing),
      optimistic: Math.round(result.optimisticEstimate.timeInMs),
      pessimistic: Math.round(result.pessimisticEstimate.timeInMs),
    }).toMatchSnapshot();
    assert.equal(result.optimisticEstimate.nodeTimings.size, 20);
    assert.equal(result.pessimisticEstimate.nodeTimings.size, 80);
    assert.ok(result.optimisticGraph, 'should have created optimistic graph');
    assert.ok(result.pessimisticGraph, 'should have created pessimistic graph');
  });

  it('should compute predicted value on iframes with substantial layout', async () => {
    const settings = {};
    const context = {settings, computedCache: new Map()};
    const result = await LanternInteractive.request({
      trace: iframeTrace,
      devtoolsLog: iframeDevtoolsLog,
      settings,
    }, context);

    expect({
      timing: Math.round(result.timing),
      optimistic: Math.round(result.optimisticEstimate.timeInMs),
      pessimistic: Math.round(result.pessimisticEstimate.timeInMs),
    }).toMatchSnapshot();
    assert.ok(result.optimisticGraph, 'should have created optimistic graph');
    assert.ok(result.pessimisticGraph, 'should have created pessimistic graph');
  });
});
