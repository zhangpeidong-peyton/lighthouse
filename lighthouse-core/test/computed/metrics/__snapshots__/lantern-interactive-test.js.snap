// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics: Lantern TTI should compute predicted value 1`] = `
Object {
  "optimistic": 4313,
  "pessimistic": 4383,
  "timing": 4348,
}
`;

exports[`Metrics: Lantern TTI should compute predicted value on iframes with substantial layout 1`] = `
Object {
  "optimistic": 6657,
  "pessimistic": 6692,
  "timing": 6674,
}
`;
